src/graph/visitor/VidExtractVisitor.cpp

```diff
@@ -135,37 +135,37 @@ void VidExtractVisitor::visit(ArithmeticExpression *expr) {
 
 void VidExtractVisitor::visit(RelationalExpression *expr) {
   if (expr->kind() == Expression::Kind::kRelIn) {
-    //    const auto *inExpr = static_cast<const RelationalExpression *>(expr);
+    // id(V) IN [List]
     if (expr->left()->kind() == Expression::Kind::kLabelAttribute) {
       const auto *labelExpr = static_cast<const LabelAttributeExpression *>(expr->left());
       const auto &label = labelExpr->left()->toString();
       vidPattern_ = VidPattern{VidPattern::Special::kInUsed,
                                {{label, {VidPattern::Vids::Kind::kOtherSource, {}}}}};
       return;
     }
+
     if (expr->left()->kind() != Expression::Kind::kFunctionCall ||
-        expr->right()->kind() != Expression::Kind::kConstant) {
+        expr->right()->kind() != Expression::Kind::kList) {
       vidPattern_ = VidPattern{};
       return;
     }
+
     const auto *fCallExpr = static_cast<const FunctionCallExpression *>(expr->left());
     if (fCallExpr->name() != "id" && fCallExpr->args()->numArgs() != 1 &&
         fCallExpr->args()->args().front()->kind() != Expression::Kind::kLabel) {
       vidPattern_ = VidPattern{};
       return;
     }
-    const auto *constExpr = static_cast<const ConstantExpression *>(expr->right());
-    if (constExpr->value().type() != Value::Type::LIST) {
-      vidPattern_ = VidPattern{};
-      return;
-    }
-    vidPattern_ = VidPattern{VidPattern::Special::kInUsed,
-                             {{fCallExpr->args()->args().front()->toString(),
-                               {VidPattern::Vids::Kind::kIn, constExpr->value().getList()}}}};
+
+    auto *listExpr = static_cast<ListExpression *>(expr->right());
+    QueryExpressionContext ctx;
+    vidPattern_ =
+        VidPattern{VidPattern::Special::kInUsed,
+                   {{fCallExpr->args()->args().front()->toString(),
+                     {VidPattern::Vids::Kind::kIn, listExpr->eval(ctx(nullptr)).getList()}}}};
```

Comment:
Maybe you should check whether the ListExpression evaluable.

Reply:
Good catch. Fixed.